src/workers/block_listenser.rs

```diff
@@ -296,94 +294,60 @@ impl BlockListener {
-                        );
-                        continue;
-                    }
+            info!("Listening to blocks {commitment_config:?}");
 
-                    if let Err(_) = this
-                        .index_slot(slot, commitment_config, postgres.clone())
-                        .await
-                    {
-                        // usually as we index all the slots even if they are not been processed we get some errors for slot
-                        // as they are not in long term storage of the rpc // we check 10 times before ignoring the slot
-                        tokio::time::sleep(tokio::time::Duration::from_millis(50)).await;
-                        {
-                            let mut queue = slots_task_queue.lock().await;
-                            queue.push_back((slot, error_count + 1));
-                        }
-                    };
-                    //   println!("{i} thread done slot {slot}");
-                }
-            });
-        }
+            let mut slot_que = Vec::new();
 
-        let rpc_client = self.rpc_client.clone();
-        tokio::spawn(async move {
-            let slots_task_queue = slots_task_queue.clone();
-            let last_latest_slot = self
-                .block_store
-                .get_latest_block_info(commitment_config)
-                .await
-                .slot;
-            // -5 for warmup
-            let mut last_latest_slot = last_latest_slot - 5;
-
-            // storage for recent slots processed
-            let rpc_client = rpc_client.clone();
             loop {
-                let new_slot = rpc_client
-                    .get_slot_with_commitment(commitment_config)
+                let mut new_block_slots = self
+                    .rpc_client
+                    .get_blocks_with_commitment(latest_slot, None, commitment_config)
                     .await?;
 
-                if last_latest_slot == new_slot {
-                    tokio::time::sleep(tokio::time::Duration::from_millis(200)).await;
-                    println!("no slots");
+                if new_block_slots.is_empty() {
+                    warn!("{latest_slot} No slots");
+                    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
                     continue;
                 }
 
-                // filter already processed slots
-                let new_block_slots: Vec<u64> = (last_latest_slot..new_slot).collect();
-                // context for lock
-                {
-                    let mut lock = slots_task_queue.lock().await;
-                    for slot in new_block_slots {
-                        lock.push_back((slot, 0));
-                    }
+                let new_latest_slot = *new_block_slots.last().unwrap();
+
+                if latest_slot == new_latest_slot {
+                    warn!("No new slots for {latest_slot}");
+                    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
```

Comment:
Reduce this wait to 200ms.

Reply:
Fixed in e61895092d12469746ed595a616e28a9dd507757